src/test/kotlin/racing/domain/strategy/StopOrStraightMoveStrategyTest.kt

```diff
@@ -0,0 +1,36 @@
+package racing.domain.strategy
+
+import io.kotest.core.spec.style.BehaviorSpec
+import io.kotest.matchers.shouldBe
+import racing.domain.DirectionType
+import racing.domain.DirectionType.BACK
+import racing.domain.DirectionType.LEFT
+import racing.domain.DirectionType.RIGHT
+import racing.domain.DirectionType.STOP
+import racing.domain.Distance
+
+class StopOrStraightMoveStrategyTest : BehaviorSpec({
+    Given("전진 혹은 멈춤 전략은") {
+        When("이동 속도가 1이고") {
+            val strategy = StopOrStraightMoveStrategy(Distance(1L))
+
+            And("이동 타입이 STRAIGHT 인 경우") {
+                val actual = strategy.move(movableStrategy = { DirectionType.STRAIGHT }, baseDistance = Distance.ZERO)
+
+                Then("이동 거리는 1만큼 증가한다") {
+                    actual shouldBe Distance(1L)
+                }
+            }
+
+            listOf(STOP, BACK, LEFT, RIGHT).forEach {
+                And("이동 타입이 $it 인 경우") {
+                    val actual = strategy.move(movableStrategy = { it }, baseDistance = Distance.ZERO)
+
+                    Then("멈춰 있는다.") {
+                        actual shouldBe Distance.ZERO
+                    }
+                }
+            }
+        }
+    }
+})
```

Comment:
Kotest는 BDD의 또 다른 방식인 DescribeSpec을 지원합니다!

DescribeSpec은 BehaviorSpec과는 달리 아래 패턴으로 구성되는데요, 
- Describe: 묘사할 대상
- Context: 주위의 환경, 상황
- It: 대상의 행동 

개인적으로 어떤 객체의 행위를 묘사하고자 할때 적합한 방식이라고 생각해요.
(Javascript 진영에서 이런 스타일을 사용한다고 하더라구요)

이를 바탕으로 `이동 전략`을 테스트하면 다음과 같이 나타낼 수 있을 것 같아요.
그냥 이런 방식도 있구나 정도로 생각해주시면 좋을 것 같습니다!
```suggestion
class StopOrStraightMoveStrategyTest : DescribeSpec({
    describe("전진 혹은 멈춤 전략은") {
        context("이동 속도가 1이고") {
            val strategy = StopOrStraightMoveStrategy(Distance(1L))

            context("이동 타입이 STRAIGHT 인 경우") {
                val actual = strategy.move(movableStrategy = { DirectionType.STRAIGHT }, baseDistance = Distance.ZERO)

                it("이동 거리를 1만큼 증가시킨다") {
                    actual shouldBe Distance(1L)
                }
            }
        }
    }
})
```